src/Swarm/Game/State.hs

```diff
@@ -515,7 +522,7 @@ playScenario em scenario userSeed toRun g = do
       & ix 0 . robotInventory
         %~ case scenario ^. scenarioCreative of
           False -> id
-          True -> const (fromElems (map (0,) things))
+          True -> union (fromElems (map (0,) things))
```

Comment:
What is the purpose of this change?

Reply:
[Fix creative mode ignoring scenario inventory](https://github.com/swarm-game/swarm/pull/400/commits/602cefa70739883a076109495df622b17cbb9383) :slightly_smiling_face:

If I add inventory to the base in a scenario, originally it would overwrite it with zero of everything.
I could workaround by adding `program: "create X; create Y; ..."`, but I thought it was a bug, so I fixed it.

Reply:
The reason why I looked for it was that the new test case would not work without inventory in creative mode. 

(I was also trying modified versions of it where you start with a logger and give it to the robot.)

I expect being able to specify inventory even in creative mode will come in handy. :slightly_smiling_face:

Reply:
Aha, I see now! 👍